via/services/url_details.py

```diff
@@ -21,19 +28,19 @@ def get_url_details(self, url, headers=None):
         :return: 2-tuple of (mime type, status code)
 
         :raise BadURL: When the URL is malformed
+        :raise checkmatelib.BadURL: if the URL is blocked by Checkmate
         :raise UpstreamServiceError: If we server gives us errors
         :raise UnhandledException: For all other request based errors
         """
-        if headers is None:
-            headers = OrderedDict()
+        if self._youtube.enabled and self._youtube.get_video_id(url):
+            return "video/x-youtube", 200
+
+        self._checkmate.raise_if_blocked(url)
 
         if GoogleDriveAPI.parse_file_url(url):
             return "application/pdf", 200
 
-        if self._youtube.enabled and self._youtube.get_video_id(url):
-            return "video/x-youtube", 200
-
-        headers = add_request_headers(clean_headers(headers))
+        headers = add_request_headers(clean_headers(headers or OrderedDict()))
```

Comment:
Merged the initialization of `headers` from above (`if headers is None: ...`) into this assignment

Reply:
With new Python, do we need `OrderedDict` any more?

Reply:
Good question. Normal dicts now keep the key insertion order since ages ago (Python 3.6) so I think that would have already been true when this code was originally written (Via was never on Python <3.6), so I'm not sure why `OrderedDict` was originally used here.

There are still some differences between dicts and `OrderedDict`'s. For example `OrderedDict`'s consider the order in equality comparisions, dicts don't. `OrderedDict.popitem()` has a new optional argument. `OrderedDict` has `move_to_end()`.

I think I'd rather not mess with it in this PR: without spending more time looking I'm not sure it won't accidentally change some behaviour